.github/workflows/release.yml

```diff
@@ -19,21 +21,25 @@ jobs:
         with:
           go-version: 1.21.x
 
+          # The default cache key for this action considers only the `go.sum` file.
+          # We include .goreleaser.yaml here to differentiate from the cache used by the push action
+          # that runs unit tests. This job produces and uses a different cache.
+          cache-dependency-path: |
+            go.sum
+            .goreleaser.yml
+
       - name: Import GPG key
         id: import_gpg
         uses: crazy-max/ghaction-import-gpg@v2
         env:
           GPG_PRIVATE_KEY: ${{ secrets.GPG_PRIVATE_KEY }}
           PASSPHRASE: ${{ secrets.PASSPHRASE }}
 
-      - name: Pull external libraries
-        run: make vendor
-
       - name: Run GoReleaser
         uses: goreleaser/goreleaser-action@v6
         with:
           version: ~> v2
-          args: release --clean
+          args: release --clean ${{ !startsWith(github.ref, 'refs/tags/v') && '--snapshot' || '' }}
```

Comment:
Also see https://docs.github.com/en/actions/learn-github-actions/expressions#example

Reply:
What is `github.ref`'s value when something is in the merge queue?

Reply:
Something like `gh-readonly-queue/main/...`, definitely not `refs/tags/...`.